lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -294,8 +308,29 @@ def write(chunk)
             end
           end
         end
+        if @running_cloudfunctions
+          # If the current group of entries is coming from a Cloud Functions
+          # function, the function name can be extracted from the tag.
+          match_data = @cloudfunctions_tag_regexp.match(tag)
+          if match_data
+            # Service name is set to Cloud Functions only for logs actually
+            # coming from a function.
+            @service_name = CLOUDFUNCTIONS_SERVICE
+            labels = write_log_entries_request['commonLabels']
+            labels["#{CLOUDFUNCTIONS_SERVICE}/function_name"] =
+              match_data['function_name']
+          else
+            # Other logs are considered as coming from the Container Engine
+            # service.
+            @service_name = CONTAINER_SERVICE
```

Comment:
I see.  Yeah, this is probably the best option for now.  I think I'm overdue for a big refactoring since we've grafted on a lot of functionality and it's getting unwieldy.